File: benchmarks/Mandel.java
public class Mandel {
    public static void main(String[] args) {
        int n = 4000; long sum = 0;
        for (int py = 0; py < n; py++) {
            double y0 = (py * 2.0 / n) - 1.0;
            for (int px = 0; px < n; px++) {
                double x0 = (px * 2.5 / n) - 2.0;
                double x = 0, y = 0; int it = 0;
                while (x*x + y*y <= 4.0 && it < 100) {
                    double xt = x*x - y*y + x0;
                    y = 2*x*y + y0; x = xt; it++;
                }
                sum += it;
            }
        }
        System.out.println(sum);
    }
}

File: benchmarks/mandel.rs
fn main() {
    let n = 4000i64; let mut sum: i64 = 0;
    for py in 0..n {
        let y0 = (py as f64) * 2.0 / (n as f64) - 1.0;
        for px in 0..n {
            let x0 = (px as f64) * 2.5 / (n as f64) - 2.0;
            let mut x = 0.0f64; let mut y = 0.0f64; let mut it = 0i64;
            while x*x + y*y <= 4.0 && it < 100 {
                let xt = x*x - y*y + x0;
                y = 2.0*x*y + y0; x = xt; it += 1;
            }
            sum += it;
        }
    }
    println!("{}", sum);
}
